src/cljs/main/broadfcui/page/workspace/notebooks/notebooks_table.cljs

```diff
@@ -333,24 +333,24 @@
            notebooks-to-localize (filter (every-pred (comp (partial = (:clusterName cluster)) :cluster-name)
                                                      (comp false? :localized?))
                                          notebooks)]
-       (endpoints/localize-notebook (:googleProject cluster) (:clusterName cluster)
-                                    (reduce merge (conj (map (partial this :-localize-entry) notebooks-to-localize) (this :-delocalize-json)))
-                                    (fn [{:keys [success? get-parsed-response]}]
-                                      (if success?
-                                        (let [updated-notebooks (map (fn [n]
-                                                                       (if (= (:cluster-name n) (:clusterName cluster))
-                                                                         (merge n {:localized? true})
-                                                                         n))
-                                                                     notebooks)]
-                                          ; flip :localized? to true so we don't re-localize notebooks to this cluster
-                                          (swap! state assoc :server-response
-                                                 {:notebooks (map (fn [n]
-                                                                    (if (= (:cluster-name n) (:clusterName cluster))
-                                                                      (merge n {:localized? true})
-                                                                      n))
-                                                                  notebooks)})
-                                          (this :-persist-notebook-cluster-association updated-notebooks))
-                                        (js/setTimeout #(this :-localize-notebooks cluster) 10000))))))
+       (when (not-empty notebooks-to-localize)
```

Comment:
With this I no longer see many `/localize` calls containing only:
```
~/workspace/.delocalize.json: "data:application/json,{"destination":"gs://fc-9573d982-0e8b-4128-bb4d-0f9d6e692bfb/notebooks","pattern":""}"
```
Before, we were localizing this file on a timer for each Running cluster. Now, we only localize it when the notebook itself is localized (which is a one-time thing when the notebook is associated to the cluster).

Reply:
Does this mean the changes a user may make won't be saved to the workspace bucket?

Reply:
No the `.delocalize.json` file still gets localized when the notebook is localized - it now just doesn't get repeatedly localized over and over again.

Reply:
Use `(when (seq collection) ...)` to test for emptiness. Or `(when-not (empty? collection) ...)`.